jax/experimental/pjit.py

```diff
@@ -803,21 +804,15 @@ def _pjit_partial_eval(trace, *in_tracers,
 
   known_ins = tuple(pv.is_known() for pv in in_pvals)
   unknown_ins = tuple(not k for k in known_ins)
-  raw_known_jaxpr, raw_unknown_jaxpr, unknown_outs = pe.partial_eval_jaxpr(
+  known_jaxpr, unknown_jaxpr, unknown_outs, res_avals = pe.partial_eval_jaxpr_nounits(
       jaxpr, unknown_ins, instantiate=False)
   unknown_outs = tuple(unknown_outs)
   known_outs = tuple(not uk for uk in unknown_outs)
-  num_residuals = len(raw_known_jaxpr.jaxpr.outvars) - len(unknown_outs)
+  num_residuals = len(res_avals)
 
   def keep_where(l, should_keep):
     return tuple(x for x, keep in zip(l, should_keep) if keep)
 
-  # Prepare the known jaxpr
-  # TODO(apaszke): map_jaxpr will break caching!
```

Comment:
Did you kill this TODO as well?

Reply:
Yep! No more `map_jaxpr`, and the jaxpr munging functions here are all cached.